formidling/src/main/java/no/nav/ung/sak/formidling/FormidlingTjeneste.java

```diff
@@ -0,0 +1,131 @@
+import no.nav.ung.sak.kontrakt.formidling.vedtaksbrev.VedtaksbrevValgDto;
+import no.nav.ung.sak.kontrakt.formidling.vedtaksbrev.VedtaksbrevValgRequestDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Dependent
+public class FormidlingTjeneste {
+
+    private BehandlingRepository behandlingRepository;
+    private BrevGenerererTjeneste brevGenerererTjeneste;
+    private VedtaksbrevRegler vedtaksbrevRegler;
+    private VedtaksbrevValgRepository vedtaksbrevValgRepository;
+
+    private static final Logger LOG = LoggerFactory.getLogger(FormidlingTjeneste.class);
+
+    @Inject
+    public FormidlingTjeneste(
+        BrevGenerererTjeneste brevGenerererTjeneste,
+        VedtaksbrevRegler vedtaksbrevRegler,
+        VedtaksbrevValgRepository vedtaksbrevValgRepository,
+        BehandlingRepository behandlingRepository) {
+        this.brevGenerererTjeneste = brevGenerererTjeneste;
+        this.vedtaksbrevRegler = vedtaksbrevRegler;
+        this.vedtaksbrevValgRepository = vedtaksbrevValgRepository;
+        this.behandlingRepository = behandlingRepository;
+    }
+
+    FormidlingTjeneste() {
+    }
+
+
+    public VedtaksbrevValgDto vedtaksbrevValg(Long behandlingId) {
+        var behandling = behandlingRepository.hentBehandling(behandlingId);
+        var erAvsluttet = behandling.erAvsluttet();
+
+        var valg = vedtaksbrevValgRepository.finnVedtakbrevValg(behandlingId);
+
+        VedtaksbrevRegelResulat resultat = vedtaksbrevRegler.kjør(behandlingId);
+        LOG.info("VedtaksbrevRegelResultat: {}", resultat.safePrint());
+
+        var egenskaper = resultat.vedtaksbrevEgenskaper();
+
+        return new VedtaksbrevValgDto(
+            egenskaper.harBrev(),
+            null,
+            false,
+            egenskaper.kanHindre(),
+            valg.map(VedtaksbrevValgEntitet::isHindret).orElse(false),
+            !erAvsluttet && egenskaper.kanOverstyreHindre(),
+            egenskaper.kanRedigere(),
+            valg.map(VedtaksbrevValgEntitet::isRedigert).orElse(false),
+            !erAvsluttet && egenskaper.kanOverstyreRediger(),
+            resultat.forklaring(),
+            valg.map(VedtaksbrevValgEntitet::getRedigertBrevHtml).orElse(null)
+        );
+    }
+
+    public boolean måSkriveBrev(Long behandlingId) {
+        var regelResulat = vedtaksbrevRegler.kjør(behandlingId).vedtaksbrevEgenskaper();
+        return regelResulat.harBrev() && regelResulat.kanRedigere() && !regelResulat.kanOverstyreRediger();
+    }
+
+    public VedtaksbrevValgEntitet lagreVedtaksbrev(VedtaksbrevValgRequestDto dto) {
+        var behandling = behandlingRepository.hentBehandling(dto.behandlingId());
+        if (behandling.erAvsluttet()) {
+            throw new BadRequestException("Kan ikke endre vedtaksbrev på avsluttet behandling");
+        }
+
+        var vedtaksbrevValgEntitet = vedtaksbrevValgRepository.finnVedtakbrevValg(dto.behandlingId())
```

Comment:
Skulle vi laget ein kopi for versjonering eller er ikkje det så viktig her?

Reply:
vi har ikke gjort det for k9-formidling. Denne vil nok bli oppdatert ganske hyppig (hver 5. sekund mens de redigerer.) så jeg tror det greit å ha det sånn.